zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -898,8 +898,16 @@ pub(crate) fn add_transparent_account_balances(
          JOIN transactions t ON t.id_tx = u.transaction_id
          -- the transaction that created the output is mined and with enough confirmations
          WHERE (
-            t.mined_height < :mempool_height -- tx is mined
-            AND :mempool_height - t.mined_height >= :min_confirmations -- has at least min_confirmations
+            -- tx is mined and has at least has at least min_confirmations
+            (
+                t.mined_height < :mempool_height -- tx is mined
+                AND :mempool_height - t.mined_height >= :min_confirmations
+            )
+            -- or the utxo can be spent with zero confirmations and is definitely unexpired
+            OR (
+                :min_confirmations = 0
+                AND (t.expiry_height = 0 OR t.expiry_height >= :mempool_height)
+            )
```

Comment:
This by comparison is fine, because the first conditional always implies `:mempool_height - t.mined_height >= 1` (the `:min_confirmations` in the second half of the first conditional just tightens that bound), and the `:min_confirmations = 0` here thus covers TXOs that are never selected by the first conditional.